pgmpy/factors/discrete/DiscreteFactor.py

```diff
@@ -710,6 +710,11 @@ def copy(self):
         # because __init__ methods does that.
         return DiscreteFactor(self.scope(), self.cardinality, self.values)
 
+    def is_valid_cpd(self):
+        return np.allclose(self.to_factor().marginalize(self.scope()[:1], inplace=False).values.flatten('C'),
+                            np.ones(np.product(self.cardinality[:0:-1])),
+                            atol=0.01)
```

Comment:
@linzhp Any particular reasons for choosing a tolerance of `0.01`. I find it somewhat low. I think tolerance would also depend on type of CPD being considered for eg some cpd can be represented discretely with better accuracy as compared to others so tolerance should be a variable quantity.